File: software_challenge_sdk/src/server_api/sc/protocol/LobbyProtocol.java
package sc.protocol;

import java.util.Arrays;
import java.util.Collection;

import sc.protocol.requests.*;
import sc.protocol.responses.*;
import sc.protocol.responses.PrepareGameProtocolMessage;
import sc.shared.*;

import com.thoughtworks.xstream.XStream;

public abstract class LobbyProtocol
{
	public static XStream registerMessages(XStream xStream)
	{
		registerAdditionalMessages(xStream, Arrays.asList(new Class<?>[] {
				ProtocolErrorMessage.class, GamePausedEvent.class,
				JoinGameProtocolMessage.class, LeftGameEvent.class,
				MementoPacket.class, PrepareGameProtocolMessage.class,
				ObservationProtocolMessage.class, RoomPacket.class }));

		registerAdditionalMessages(xStream, Arrays.asList(new Class<?>[] {
				AuthenticateRequest.class, CancelRequest.class,
				FreeReservationRequest.class, JoinPreparedRoomRequest.class,
				JoinRoomRequest.class, ObservationRequest.class,
				PauseGameRequest.class, ControlTimeoutRequest.class, PrepareGameRequest.class,
				StepRequest.class, GetScoreForPlayerRequest.class, TestModeRequest.class,
				PlayerScorePacket.class, TestModeMessage.class, GameRoomMessage.class
		}));
		registerAdditionalMessages(xStream, Arrays.asList(new Class<?>[] {
				GameResult.class, PlayerScore.class, ScoreAggregation.class, PlayerColor.class,
				ScoreCause.class, ScoreDefinition.class, ScoreFragment.class, WinCondition.class,
				SlotDescriptor.class, Score.class, ScoreValue.class}));

		return xStream;
	}

	public static XStream registerAdditionalMessages(XStream xstream,
			Collection<Class<?>> protocolClasses)
	{
		if(protocolClasses != null)
		{
			for (Class<?> clazz : protocolClasses)
			{
				xstream.processAnnotations(clazz);
			}
		}

		return xstream;
	}
}

File: sdk/src/server-api/sc/protocol/helpers/LobbyProtocol.kt
package sc.protocol.helpers

import com.thoughtworks.xstream.XStream
import sc.api.plugins.ITeam
import sc.protocol.requests.*
import sc.protocol.responses.*
import sc.shared.*

object LobbyProtocol {

    @JvmStatic
    fun registerMessages(xStream: XStream): XStream {
        registerAdditionalMessages(xStream, listOf(ProtocolErrorMessage::class.java, GamePausedEvent::class.java, JoinGameProtocolMessage::class.java, LeftGameEvent::class.java, MementoPacket::class.java, PrepareGameProtocolMessage::class.java, ObservationProtocolMessage::class.java, RoomPacket::class.java, WelcomeMessage::class.java))
    
        registerAdditionalMessages(xStream, listOf(AuthenticateRequest::class.java, CancelRequest::class.java, FreeReservationRequest::class.java, JoinPreparedRoomRequest::class.java, JoinRoomRequest::class.java, ObservationRequest::class.java, PauseGameRequest::class.java, ControlTimeoutRequest::class.java, PrepareGameRequest::class.java, StepRequest::class.java, GetScoreForPlayerRequest::class.java, TestModeRequest::class.java, PlayerScorePacket::class.java, TestModeMessage::class.java, GameRoomMessage::class.java))
    
        registerAdditionalMessages(xStream, listOf(GameResult::class.java, PlayerScore::class.java, ScoreAggregation::class.java, ITeam::class.java,
                ScoreCause::class.java, ScoreDefinition::class.java, ScoreFragment::class.java, WinCondition::class.java,
                SlotDescriptor::class.java, Score::class.java, ScoreValue::class.java))

        return xStream
    }

    @JvmStatic
    fun registerAdditionalMessages(xstream: XStream, protocolClasses: Collection<Class<*>>?): XStream {
        if(protocolClasses != null) {
            for(clazz in protocolClasses) {
                xstream.processAnnotations(clazz)
            }
        }
        return xstream
    }

}
